liblangutil/Scanner.cpp

```diff
@@ -280,6 +280,30 @@ Token Scanner::skipSingleLineComment()
 	return Token::Whitespace;
 }
 
+bool Scanner::isLineTerminator() const
+{
+	return m_char == '\n' || (m_char == '\r' && !m_source->isPastEndOfInput(2) && m_source->get(1) == '\n');
+}
+
+bool Scanner::tryScanLineTerminator()
+{
+	if (m_char == '\n')
+	{
+		advance();
+		return true;
+	}
+
+	if (m_char == '\r')
+	{
+		advance();
+		if (!m_source->isPastEndOfInput(1) && m_source->get(0) == '\n')
```

Comment:
Hm... what happens if there is no ``\n`` after the ``\r``? How do we want do deal with that? Is it OK to just treat a single ``\r`` as a regular line break (that's what's happening now, right?)?

Reply:
we compared to VSCode as source of inspiration. How it is treated:
* `\n` is newline
* `\r` is newline
* `\r\n` is newline,
that is, if there's no following `\n` to the `\r` then it's still treated as newline.

Reply:
Sounds good!